### ISSUE TITLE
dvc add --no-commit for empty file is added to cache

### ISSUE THREAD

**Issue by robogeek** [iterative/dvc#1666]
**Please provide information about your setup**
DVC version(i.e. `dvc --version`), Platform and method of installation (pip, homebrew, pkg Mac, exe (Windows), DEB(Linux), RPM(Linux))

```
$ dvc --version
0.29.0
```

Installed using pip on a macOS X machine

If I use `dvc add --no-commit` on an empty file, the file is added to the cache, while if the file has content it is not added to the cache.

```shell
$ touch nc.txt
$ dvc add --no-commit nc.txt
Warning: file/directory 'nc.txt' is empty.
Adding 'nc.txt' to '.gitignore'.
Saving information to 'nc.txt.dvc'.

To track the changes with git run:

    git add .gitignore nc.txt.dvc
$ cat nc.txt.dvc 
md5: 3d4530f1c05c176a837df864443b4b08
outs:
- cache: true
  md5: d41d8cd98f00b204e9800998ecf8427e
  metric: false
  path: nc.txt
$ ls .dvc/cache/d4
1d8cd98f00b204e9800998ecf8427e
$ cat .dvc/cache/d4/1d8cd98f00b204e9800998ecf8427e 
```

The corresponding file to the empty file is in the cache.

After seeing this behavior I wanted to try with a file that contains some data.

```shell
$ echo 'hi there' >nc2.txt
$ dvc add --no-commit nc2.txt
Adding 'nc2.txt' to '.gitignore'.
Saving information to 'nc2.txt.dvc'.

To track the changes with git run:

    git add .gitignore nc2.txt.dvc
$ cat nc2.txt.dvc 
md5: bc71e93f57271a786cc1dc6e8d157b6d
outs:
- cache: true
  md5: 12f6bb1941df66b8f138a446d4e8670c
  metric: false
  path: nc2.txt
$ ls .dvc/cache
02    1b    35    40    42    78    d4    f5
$ ls .dvc/cache/*/f6*
ls: .dvc/cache/*/f6*: No such file or directory
```

In other words, the directory `.dvc/cache/12` does not exist, which would be required to exist to hold this file in the cache.  Therefore the corresponding file is not added to the cache.

**Comment by efiop** [comment]
@robogeek I think you just had an empty file added previously :slightly_smiling_face: Here is a reproducer script:
```
#!/bin/bash              
                         
set -e                   
set -x                   
                         
rm -rf myrepo            
mkdir myrepo             
cd myrepo                
                         
git init                 
dvc init                 
                         
touch empty              
dvc add empty --no-commit
                         
tree .dvc/cache          

```

which shows that `.dvc/cache` is empty, so empty files are not added to cache in --no-commit.

**Comment by efiop** [comment]
Sorry, closed accidentally. Could you try running that script and see if it works for you?

**Comment by robogeek** [comment]
This was found to not be an issue

Namely I had previously added an empty file.  The hash code for an empty file is the same because the contents are the same.  Therefore the existing empty file was what was in the cache.
